control-plane/connect-inject/controllers/endpoints/endpoints_controller.go

```diff
@@ -999,8 +1008,87 @@ func (r *Controller) deregisterService(apiClient *api.Client, k8sSvcName, k8sSvc
 		}
 	}
 
-	return errs
+	if requeueAfter > 0 {
+		r.Log.Info("re-queueing event", "requeueAfter", requeueAfter)
```

Comment:
~ It might help to call out why we're requeueing so operators know it's expected, and due to graceful shutdown